stl/inc/xstring

```diff
@@ -3162,38 +3089,31 @@ private:
                 }
 #endif // _ITERATOR_DEBUG_LEVEL != 0
 
-#ifdef _INSERT_STRING_ANNOTATION
-                if (!_Right_data._Large_string_engaged()) {
-                    _ASAN_STRING_REMOVE(_Right);
-                }
-#endif // _INSERT_STRING_ANNOTATION
-
                 _Memcpy_val_from(_Right);
-
-#ifdef _INSERT_STRING_ANNOTATION
-                if (!_Right_data._Large_string_engaged()) {
-                    _ASAN_STRING_REMOVE(_Right);
-                    _ASAN_STRING_CREATE(*this);
-                }
-#endif // _INSERT_STRING_ANNOTATION
-
                 _Right._Tidy_init();
                 return;
             }
         }
+#endif // !defined(_INSERT_STRING_ANNOTATION)
 
         if (_Right_data._Large_string_engaged()) { // steal buffer
             _Construct_in_place(_My_data._Bx._Ptr, _Right_data._Bx._Ptr);
             _Right_data._Bx._Ptr = nullptr;
+
+            _My_data._Myres  = _Right_data._Myres;
+            _My_data._Mysize = _Right_data._Mysize;
+
             _Swap_proxy_and_iterators(_Right);
         } else { // copy small string buffer
             _My_data._Activate_SSO_buffer();
+
             _Traits::copy(_My_data._Bx._Buf, _Right_data._Bx._Buf, _Right_data._Mysize + 1);
+            _My_data._Myres  = _BUF_SIZE - 1;
+            _My_data._Mysize = _Right_data._Mysize;
+
             _Right_data._Orphan_all();
         }
 
-        _My_data._Mysize = _Right_data._Mysize;
-        _My_data._Myres  = _Right_data._Myres;

```

Comment:
I'm unsure as to why these lines are being changed. (When `_Right_data` is small, `_Right_data._Myres` should be `_BUF_SIZE - 1` by definition.)

I believe that both forms are correct, but the original form probably has slightly better codegen.